modules/react/badge/lib/CountBadge.tsx

```diff
@@ -32,35 +15,79 @@ const grow = keyframes`
   }
 `;
 
-const Container = styled('span')<CountBadgeProps>(
-  {
-    alignItems: 'center',
-    animation: `${grow} 0.2s ease`,
-    borderRadius: borderRadius.circle,
-    boxSizing: 'border-box',
-    display: 'inline-flex',
-    fontFamily,
-    fontSize: '12px',
-    fontWeight: 700,
-    height: '20px',
-    justifyContent: 'center',
-    lineHeight: '20px',
-    minWidth: '20px',
-    padding: '0 6.5px',
-  },
-  ({variant = 'default'}) => ({
-    ...variants[variant],
-  })
-);
+export interface CountBadgeProps extends CSProps {
+  /**
+   * Sets the count displayed in the badge
+   *
+   * @default 0
+   */
+  count?: number;
+  /**
+   * Sets the maximum count to display before formatting the number.
+   * E.g. Given a count of `100` and a limit of `100`, the badge would display `99+`.
+   *
+   * @default 1000
+   */
+  limit?: number;
+  /**
+   * Sets the variant of the Count Badge
+   *
+   * @default 'default'
+   */
+  variant?: 'default' | 'inverse';
+}
 
-export const CountBadge = (props: CountBadgeProps) => {
-  const {variant = 'default', count = 0, limit = 1000, ...elemProps} = props;
+function px2rem(px: number) {
+  return `${px / 16}rem`;
+}
+
+const baseStyles = createStyles({
+  alignItems: 'center',
+  animation: `${grow} 0.2s ease`,
+  borderRadius: system.shape.round,
+  boxSizing: 'border-box',
+  display: 'inline-flex',
+  fontFamily: system.fontFamily.default,
+  fontSize: system.fontSize.subtext.medium,
+  fontWeight: 700, // should use system.fontWeight.bold
```

Comment:
do we have an issue created for fontStyles not being recognized by create Styles?